ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -163,13 +181,15 @@ object Queries {
   }
 
   private[http] def selectContracts(
+      party: String,
       tpid: SurrogateTpId,
-      predicate: Fragment): Query0[DBContract[SurrogateTpId, JsValue, JsValue]] = {
-    val q = sql"""SELECT (contract_id, create_arguments, witness_parties)
+      predicate: Fragment): Query0[DBContract[Unit, JsValue, Unit]] = {
```

Comment:
is that something WIP?
what do you plan to do with a contract that does not contain `templateId` value? It's type arg is `Unit`.

Reply:
actually thinking about it... we can look up `templateId` by `contractId`, but why bother if we can fetch it from DB. Just trying to understand what is the use case for this function, don't think it is being used anywhere.

Reply:
> don't think it is being used anywhere

It's not and won't in this PR; it will be used for #3388.

> what do you plan to do with a contract that does not contain templateId value?

The use of `Unit` here represents the fact that nothing useful can be put in that slot. Logically, the template ID is always the template ID that was passed as the `tpid` argument, but `tpid.type` will stop compiling at some point in the future, so I don't want to give the false impression that any template IDs are being "looked up". The caller can insert the `tpid` afterwards if they want by `copy`, but I suspect that will never be useful in practice.